engine/class_modules/sc_death_knight.cpp

```diff
@@ -6925,14 +6944,33 @@ unsigned death_knight_t::replenish_rune( unsigned n, gain_t* gain )
 }
 
 // Helper function to trigger Killing Machine, whether it's from a "forced" proc, a % chance, or the regular rppm proc
-void death_knight_t::trigger_killing_machine( double chance, proc_t* proc, proc_t* wasted_proc )
+void death_knight_t::trigger_killing_machine( double chance, proc_t* proc, proc_t* wasted_proc, weapon_e weapon_type )
```

Comment:
You don't need to pass the player's weapon type from the triggering actions to death_knight_t::trigger_killing_machine(), it should be accessible just fine within it.